tests/test/subtype.rs

```diff
@@ -72,20 +72,49 @@ fn ref_lifetime_variance() {
 }
 
 #[test]
-fn fn_lifetime_variance() {
+fn fn_lifetime_variance_args() {
     test! {
         program {
         }
 
         goal {
-            Subtype(for<'a, 'b> fn(&'a u32, &'b u32), for<'a> fn(&'a u32, &'a u32))
+            for<'a, 'b> fn(&'a u32, &'b u32) = for<'a> fn(&'a u32, &'a u32)
         } yields {
+            "Unique;substitution [], lifetime constraints [\
+            InEnvironment { environment: Env([]), goal: '!1_0: '!1_1 }, \
+            InEnvironment { environment: Env([]), goal: '!1_1: '!1_0 }]"
+        }
+    }
+}
+
+#[test]
+fn fn_lifetime_variance_with_return_type() {
+    test! {
+        program {
+        }
+
+        goal {
+            Subtype(for<'a, 'b> fn(&'a u32, &'b u32) -> &'a u32, for<'a> fn(&'a u32, &'a u32) -> &'a u32)
+        } yields {
+            // TODO: are these results actually correct?
             "Unique; for<?U1,?U1> { substitution [], lifetime constraints [\
-            InEnvironment { environment: Env([]), goal: '!1_0: '^0.0 }, \
-            InEnvironment { environment: Env([]), goal: '!1_0: '^0.1 }] }"
+                InEnvironment { environment: Env([]), goal: '!1_0: '^0.0 }, \
+                InEnvironment { environment: Env([]), goal: '!1_0: '^0.1 }, \
+                InEnvironment { environment: Env([]), goal: '^0.0: '!1_0 }  \
```

Comment:
These are...weird. I would have expected something like: `'!1_0: '^0.0, '!1_1: '1_0, '!1_0: '1_1`.

The constraints here seem...backward?